logic/tool/TypeResolver.java

```diff
@@ -373,88 +382,72 @@ private void convertValue(TypeVariable owner, ValueConstraint<?> constraint) {
-                } else if (constraint.isPlays()) {
-                    addNeighbours(fromCopy, constraint.asPlays().role());
-                } else if (constraint.isRelates()) {
-                    addNeighbours(fromCopy, constraint.asRelates().role());
-                } else if (constraint.isIs()) {
-                    addNeighbours(fromCopy, constraint.asIs().variable());
-                }
+        private void copyNeighbours(TypeVariable from, TypeVariable to) { // TODO: why is 'to' variable never used?
+            for (TypeConstraint constraint : from.constraints()) {
+                if (constraint.isSub()) addNeighbours(from, constraint.asSub().type());
+                else if (constraint.isOwns()) addNeighbours(from, constraint.asOwns().attribute());
+                else if (constraint.isPlays()) addNeighbours(from, constraint.asPlays().role());
+                else if (constraint.isRelates()) addNeighbours(from, constraint.asRelates().role());
+                else if (constraint.isIs()) addNeighbours(from, constraint.asIs().variable());
+                // TODO: There are other constraints in a TypeVariable. Are we intentionally ignoring them?
+                //       Why do we not throw GraknException.of(ILLEGAL_STATE); ?
             }
         }
-
     }
 
-    private static class ResolverVariables {
+    private static class Resolvers {
 
-        private final Map<Identifier, TypeVariable> resolveeVars;
-        private final Map<RelationConstraint.RolePlayer, TypeVariable> rolePlayerResolveVars;
+        private final Map<Identifier, TypeVariable> typeResolvers;
+        private final Map<RelationConstraint.RolePlayer, TypeVariable> roleTypeResolvers;
+        // TODO: Why is there no checkers and getters for 'roleTypeResolvers'?
+        // TODO: do we still need 'roleTypeResolvers' given that we no longer put resolveRoleTypes on RelationConstraints?
 
-        private Integer tempCounter;
+        private int sysVarCounter;
 
-        ResolverVariables() {
-            this.tempCounter = 0;
-            this.resolveeVars = new HashMap<>();
-            this.rolePlayerResolveVars = new HashMap<>();
+        Resolvers() {
+            this.sysVarCounter = 0;
+            this.typeResolvers = new HashMap<>();
+            this.roleTypeResolvers = new HashMap<>();
         }
 
-        public TypeVariable convert(RelationConstraint.RolePlayer key) {
-            if (!rolePlayerResolveVars.containsKey(key)) {
-                TypeVariable newTypeVar = new TypeVariable(Identifier.Variable.of(
-                        new SystemReference("temp" + addAndGetCounter())));
-                rolePlayerResolveVars.put(key, newTypeVar);
-            }
-            return rolePlayerResolveVars.get(key);
+        TypeVariable register(RelationConstraint.RolePlayer rolePlayer) {
+            return roleTypeResolvers.computeIfAbsent(rolePlayer, rp -> new TypeVariable(newSystemVariable()));
```

Comment:
The entire logic in the method body was equivalent to a `computeIfAbsent()` - we should simplify code as much as possible - the shorter the better, so we do not burden the reader of the code with any unnecessary complexity in interpreting the code.